File: lemur-libs/lemur-testkit/src/main/java/io/github/malczuuu/lemur/testkit/container/PostgresAwareTest.java
package io.github.malczuuu.lemur.testkit.container;

import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
import org.testcontainers.junit.jupiter.Container;
import org.testcontainers.postgresql.PostgreSQLContainer;

/**
 * Interface to be implemented by tests that require a PostgreSQL container. It provides a
 * pre-configured PostgreSQL container for testing purposes.
 */
public interface PostgresAwareTest {

  /**
   * <strong>Must not be accessed directly by test code.</strong> To be used by test framework only.
   */
  @Container @ServiceConnection
  PostgreSQLContainer postgresContainer = new PostgreSQLContainer("postgres:18.3-alpine");
}

File: lemur-libs/lemur-testkit/src/main/kotlin/io/github/malczuuu/lemur/testkit/container/PostgresAwareTest.kt
package io.github.malczuuu.lemur.testkit.container

import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.testcontainers.junit.jupiter.Container
import org.testcontainers.postgresql.PostgreSQLContainer

interface PostgresAwareTest {

    companion object {
        @Container
        @ServiceConnection
        @JvmField
        val postgresContainer: PostgreSQLContainer = PostgreSQLContainer("postgres:18.3-alpine")
    }
}
